source/application/AMPIC887DriverServer/AMGCS2CommandFactory.cpp

```diff
@@ -0,0 +1,497 @@
+	} else if(commandString.startsWith("MVR")) {
+		return new AMGCS2MoveRelativeCommand(axesDoublePairFromCommandString(commandString));
+	} else if(commandString.startsWith("NLM?")) {
+		return new AMGCS2GetLowSoftLimitsCommand(axesFromCommandString(commandString));
+	} else if(commandString.startsWith("NLM")) {
+		return new AMGCS2SetLowSoftLimitsCommand(axesDoublePairFromCommandString(commandString));
+	} else if(commandString.startsWith("ONT?")) {
+		return new AMGCS2GetOnTargetStateCommand(axesFromCommandString(commandString));
+	} else if(commandString.startsWith("PLM?")) {
+		return new AMGCS2GetHighSoftLimitsCommand(axesFromCommandString(commandString));
+	} else if(commandString.startsWith("PLM")) {
+		return new AMGCS2SetHighSoftLimitsCommand(axesDoublePairFromCommandString(commandString));
+	} else if(commandString.startsWith("PUN?")) {
+		return new AMGCS2GetPositionUnitsCommand(axesFromCommandString(commandString));
+	} else if(commandString.startsWith("SCT?")) {
+		return new AMGCS2GetCycleTimeCommand();
+	} else if(commandString.startsWith("SCT")) {
+		return buildSetCycleTimeCommand(commandArguments(commandString));
+	} else if(commandString.startsWith("SPI?")) {
+		return new AMGCS2GetPivotPointCommand(axesFromCommandString(commandString));
+	} else if(commandString.startsWith("SPI")) {
+		return new AMGCS2SetPivotPointCommand(axesDoublePairFromCommandString(commandString));
+	} else if(commandString.startsWith("SSL?")) {
+		return new AMGCS2GetSoftLimitsStatusCommand(axesFromCommandString(commandString));
+	} else if(commandString.startsWith("SSL")) {
+		return buildSetSoftLimitStatusesCommand(commandArguments(commandString));
+	} else if(commandString.startsWith("SVO?")) {
+		return new AMGCS2GetServoModeCommand(axesFromCommandString(commandString));
+	} else if(commandString.startsWith("SVO")) {
+		return buildSetServoModeCommand(commandArguments(commandString));
+	} else if(commandString.startsWith("SST?")) {
+		return new AMGCS2GetStepSizeCommand(axesFromCommandString(commandString));
+	} else if(commandString.startsWith("SST")) {
+		return new AMGCS2SetStepSizeCommand(axesDoublePairFromCommandString(commandString));
+	} else if(commandString.startsWith("TMN?")) {
+		return new AMGCS2GetMinCommandablePositionCommand(axesFromCommandString(commandString));
+	} else if(commandString.startsWith("TMX?")) {
+		return new AMGCS2GetMaxCommandablePositionCommand(axesFromCommandString(commandString));
+	} else if(commandString.startsWith("VLS?")) {
+		return new AMGCS2GetSystemVelocityCommand();
+	} else if(commandString.startsWith("VLS")) {
+		return buildSetSystemVelocityCommand(commandArguments(commandString));
+	} else if(commandString.startsWith("VMO?")) {
+		return new AMGCS2VirtualMoveCommand(axesDoublePairFromCommandString(commandString));
+	}
+
+	return 0;
+}
+
+QStringList AMGCS2CommandFactory::commandArguments(const QString& commandString)
+{
+	QStringList argumentsList = commandString.trimmed().split(" ");
```

Comment:
For information purposes only, you could also replace these three lines with:
`return commandString.trimmed().split(" ").mid(1, -1);`